chain/network/src/peer_manager.rs

```diff
@@ -384,52 +527,83 @@ impl PeerManagerActor {
             .spawn(ctx);
     }
 
-    fn announce_account(
-        &mut self,
-        ctx: &mut Context<Self>,
-        mut announce_account: AnnounceAccount,
-        force: bool,
-    ) {
-        // If this is a new account send an announcement to random set of peers.
-        if self.routing_table.update(&announce_account).is_new() || force {
-            if announce_account.peer_id_sender() != self.peer_id {
-                // If this announcement was not sent by this peer, add peer information
-                assert!(!force);
-                announce_account.extend(self.peer_id, &self.config.secret_key);
-            }
-
+    fn announce_account(&mut self, ctx: &mut Context<Self>, mut announce_account: AnnounceAccount) {
+        // If this is an announcement of our account id.
+        if announce_account.peer_id() == self.peer_id {
```

Comment:
`.peer_id` is the last peer who sends the announcement, the peer that is being announced is in the header.
It still works (since we would never have ourselves as the *last* peer in something we just received), but I think checking the header would be more semantically correct, or the comment above needs to refelect the semantics better (i.e. explain why checking the last peer in the routing table works)